networker/networker.go

```diff
@@ -261,16 +261,19 @@ func (netw *Combined) start(
 		return err
 	}
 
-	if !netw.isMeshnetSet {
-		netw.publisher.Publish("Setting the routing rules up")
-		err = netw.policyRouter.SetupRoutingRules(
-			netw.vpnet.Tun().Interface(),
-			serverData.IP.Is6(),
-		)
+	netw.publisher.Publish("Setting the routing rules up")
 
-		if err != nil {
-			return err
-		}
+	// need to re-set routing setup
+	if err = netw.policyRouter.CleanupRouting(); err != nil {
```

Comment:
There is still a problem with LAN Discovery setup - its decision and setup is done early i.e. on cli command (set on/off, add allowlist) because LAN discovery is about local (private) addresses be enabled to access when exit routing is enforced. But also that is related very much with Allowlist - need to merge addresses. But when we have to make decision about Route LAN access to remote peer or not - here we need to disable LAN discovery in case remote peer allows its LAN access, but LAN Discovery setup is already blended/mixed with Allowlist setup and we have no clear way to manage LAN Discovery setting without risk screwing Allowlist setup. IMO, routing setup decisions has to be done as late as possible or need possibility to re-play LAN Discovery and Allowlist setup.

Reply:
short term solution: disable lan-discovery when mesh peer connected;
long term solution: design and implement networker state machine;